sort.php

```diff
@@ -23,6 +23,50 @@ function checkForEquality($firstNum, $secondNum)
     return $firstNum < $secondNum;
 }
 
+function swapElement(&$firstParam, &$secondParam) {
+    $temp = $firstParam;
+    $firstParam = $secondParam;
+    $secondParam = $temp;
+}
+
+function partition(&$arr, $lowest, $highest) {
+    $pivot = $arr[$highest];
+    $i = $lowest;
+    $j = $highest - 1;
+
+    while ($i < $j) {
+        while ($i < $highest && $arr[$i] < $pivot) {
+            $i += 1;
+        }
+
+        while ($j > $lowest && $arr[$j] >= $pivot) {
+            $j -= 1;
+        }
+
+        if ($i < $j) {
+            swapElement($arr[$i], $arr[$j]);
+        }
+    }
+
+    if ($arr[$i] > $pivot) {
+        swapElement($arr[$i], $arr[$highest]);
+    }
+
+    return $i;
+}
+
+function quickSort(&$arr, $lowest, $highest)
```

Comment:
Why do you pass $arr by reference here?

Reply:
I tried it out beforehand without returning the array and when it worked I forgot to put ``&`` it away.